### PR TITLE
[Doc][Process] sweep agent-loaded rules/ + domain-rules/ for brevity

### LINKED ISSUE #1277: [Docs][Process] sweep agent-loaded docs for brevity (rules / domain-rules / SKILLs / design)
## Description

Per project feedback `feedback_agent_doc_brevity.md`: every doc that an agent loads at runtime must be maximally brief; design docs hold top-level decisions agents reference, not stale-prone implementation details. This sweep applies that standard to all agent-loaded docs in the repo.

## Goal

Reduce verbosity in agent-loaded docs without losing actual rules. Apply: one imperative sentence per directive, no narrative throat-clearing, no design-vs-checklist duplication, no implementation details that age out.

## Plan

- Pass 1: \`.claude/rules/*.md\` (high-frequency: agent loads on every invocation).
- Pass 2: \`.claude/domain-rules/*.md\` (load-on-demand by file pattern).
- Pass 3: \`.claude/skills/*/SKILL.md\` (loaded when skill is invoked; ~1700 lines total).
- Pass 4: \`docs/design/*.md\` cross-referenced by the above (~2000 lines; cut implementation details, keep decisions).

For each file: cut sentences that don't carry a directive or decision. Move implementation details to code/commit/PR. Replace design-vs-checklist duplication with \`See: ...\` cross-references.

## Constraints

- Docs-only PR. No code / manifest / test / benchmark changes.
- Don't drop actual rules — only restate them tighter.
- Preserve all existing cross-references; just shorten the prose around them.
- Manifest is byte-identical (this is a process/docs PR).

## Acceptance Criteria

- [ ] Net line reduction across the four passes (target ~30%+ per file where applicable).
- [ ] No actual rule lost — for each removed sentence, the rule it carried is still expressed somewhere (either in the same file or via cross-reference).
- [ ] No design-vs-checklist duplication remains in touched files.
- [ ] \`pre-commit run --files <changed docs>\` passes; markdown lint clean.

### PR DESCRIPTION
**Pull request by lcy-seso**
## Summary

Per `feedback_agent_doc_brevity.md`: agents load these docs on every invocation; verbose phrasing burns context budget and buries the rule. This PR is **Pass 1 + Pass 2** of the brevity sweep tracked in #1277.

- `.claude/rules/code-style.md` — collapse FIXME(staged-rollout) preamble; tighten Google-docstring, abbreviation, and dev-process-metadata bullets.
- `.claude/domain-rules/manifest-spec.md` — merge near-duplicate bullets; collapse comment-policy preamble.
- `.claude/domain-rules/{benchmark,ops-design,testing-budget}.md` — one imperative sentence per directive.

`security.md`, `manifest-trust-model.md`, `manifest-validator.md`, `design-docs.md` left as-is — already at one-bullet-per-rule.

## Test plan

- [x] No rule lost: every removed sentence's directive is still expressed (verified by hand against the pre-sweep diff).
- [x] Cross-references preserved.
- [x] `pre-commit run --files <changed docs>` clean (mdformat round-trip applied).

## Scope notes

Pass 3 (`.claude/skills/*/SKILL.md`) and Pass 4 (`docs/design/*.md`) deferred to separate PRs. SKILL files are operational protocols loaded only on skill invocation (lower context cost), and tightening them risks dropping load-bearing detail sub-skills depend on. Design docs need per-file judgment about which sections are decisions vs implementation snapshots.

Closes tile-ai/TileOPs#1277

### CODE REVIEW

**Review by Ibuki-wind — DISMISSED** [review]

File: .claude/rules/code-style.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
-- Every `tileops/kernels/*` subpackage must have an `__init__.py` with explicit `__all__` and `from .module import Symbol` re-exports.
+- Every `tileops/kernels/*` subpackage has an `__init__.py` with explicit `__all__` and `from .module import Symbol` re-exports.
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/rules/code-style.md]: Fixed: re-added 'MUST' to make this a directive rather than a description.

File: .claude/domain-rules/manifest-spec.md [hunk 2]
```diff
@@ -8,42 +8,42 @@
 
 ______________________________________________________________________
 
-- Manifest key must equal the corresponding Op `cls.__name__` exactly. Class-naming convention is in `ops-design.md`.
+- Manifest key must equal the Op `cls.__name__` exactly.
 
-- `ref_api` (required) — declares the external API the signature follows (e.g., `torch.nn.functional.rms_norm`). Set to `"none"` if no direct external counterpart exists. Informational only; does not affect validation.
+- `ref_api` (required): the external API the signature mirrors (e.g. `torch.nn.functional.rms_norm`); `"none"` if none. Informational, not validated.
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/domain-rules/manifest-spec.md]: Fixed: clarified that the validator enforces presence + string type but not semantics — 'not validated' was misleading.

File: .claude/domain-rules/testing-budget.md [hunk 3]
```diff
@@ -7,10 +7,10 @@
...
-- Run `scripts/test_node_delta.py` before submitting PRs that modify test files. No growth on existing files: nothing to report. Growth on existing files: include script output and one-line justification in PR description. New test files only: no delta report needed.
-- Binary operator tests must cover broadcast semantics: bias-add `(B,S,D)+(1,1,D)`, row `(B,S,D)+(B,S,1)`, scalar `(M,N)+(1,1)`. Applies to arithmetic, comparison, logical, and bitwise binary ops.
-- Skill development tests stay local — do not commit anything under `.claude/skills/`.
+- Every test case traces to a specific code path, dtype dispatch, or regression. No cases for combinatorial confidence.
+- Test all supported dtypes. Don't cross dtype and shape coverage unless the combination triggers a distinct code path.
+- Don't generate fixtures from `tileops/manifest/` workloads. Test parameters are a curated correctness subset.
+- Before committing: drop scaffolding tests that guarded intermediate implementation steps but no final code path.
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/domain-rules/testing-budget.md]: Fixed: rewrote as 'don't guard any final code path' to remove the missing-verb ambiguity.

File: .claude/domain-rules/ops-design.md [hunk 4]
```diff
@@ -7,10 +7,10 @@
...
-- When adding or modifying an intermediate base class, changing kernel dispatch patterns, or introducing new class variable protocols, update `docs/design/ops-design.md` to reflect the change.
-- When adding a new op family that inherits `Op` directly, evaluate whether it shares `forward()` flow with an existing family before creating a new base class. Document the decision in the PR.
+- Op class names: PascalCase `{Name}{Direction}Op`, direction suffix mandatory. Manifest author chooses `{Name}`.
+- Kernel class names: PascalCase `{Name}{Direction}Kernel`. Builder functions stay snake_case.
+- `kernel_map` is the Op→Kernel dispatch registration table: snake_case dispatch keys (decoupled from class names) → Kernel class names. Manifest declares it; agents implement the listed Kernels. See [ops-design-reference.md § Kernel Dispatch](../../docs/design/ops-design-reference.md#kernel-dispatch-kernel_map).
+- Op `__init__` is keyword-only (`def __init__(self, *, ...)`). Parameter names come from the manifest: `shape` dim names (fixed-rank), `static_dims` keys (arbitrary-rank), `params` keys. Nothing not in the manifest belongs in `__init__`.
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/domain-rules/ops-design.md]: Fixed: rewrote in positive form ('Only manifest-declared information belongs in __init__').

File: .claude/domain-rules/manifest-spec.md [hunk 5]
```diff
@@ -8,42 +8,42 @@
...
+- `inputs`, `outputs`, `params` are ordered dicts — key order is signature position. Don't reorder.
 
-- Params include all PyTorch-supported parameters, even if the current kernel only supports the default. Params default to `__init__` kwargs (architecture-decided, fixed for the Op instance's lifetime); in rare cases a param belongs in `forward()` when PyTorch's reference API requires it or when the value is per-batch — justify the exception in the op's introducing issue.
+- Op signatures must match PyTorch's public API (parameter names, set, semantics). Don't invent parameters.
 
-- `dtype` syntax: `|` for alternatives. `same_as(ref)` is a dtype-only identity constraint: the tensor must have the exact same dtype as `ref` at runtime, does not contribute an independent axis to the Cartesian product in `dtype_combos`, and must not be used for shape.
+- Params include all PyTorch-supported parameters even if the kernel only honors the default. Default to `__init__` kwargs (architecture-decided, lifetime-fixed); a param belongs in `forward()` only when the reference API requires it or the value is per-batch — justify in the introducing issue.
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/domain-rules/manifest-spec.md]: Applied: merged 'Op signatures match PyTorch' with the params-completeness bullet into a single PyTorch-API-alignment directive.

File: .claude/domain-rules/manifest-spec.md [hunk 6]
```diff
@@ -8,42 +8,42 @@
...
+- `dtype` syntax: `|` for alternatives. `same_as(ref)` is dtype-only identity (matches `ref` at runtime, no extra axis in `dtype_combos`, never used for shape).
 
-- Every output tensor's shape must be fully specified via `shape` and/or `shape_rules`. Inputs may omit `shape` (→ arbitrary rank).
+- `dtype_combos` only when the supported set is a strict subset of the Cartesian product. Omit when all combinations are valid.
 
-- `shape` present = fixed rank. Names become roofline variables. `shape` absent = arbitrary rank, use `params` + `shape_rules`.
+- Every output tensor's shape is fully specified by `shape` and/or `shape_rules`. Inputs may omit `shape` (= arbitrary rank).
 
-- Shared dimension names across tensors = sizes must match.
+- `shape` present → fixed rank, names become roofline variables. `shape` absent → arbitrary rank; use `params` + `shape_rules`.
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/domain-rules/manifest-spec.md]: Applied: merged the shape-fixed-rank and shape-arbitrary-rank bullets into one rank-specification bullet (also folded the shared-dim-name rule).

File: .claude/domain-rules/manifest-spec.md [hunk 7]
```diff
@@ -8,42 +8,42 @@
...
 
-- `shape_rules` are Python expressions for shape relationships. `shape` and `shape_rules` fully specify output shape derivation.
+- Shared dimension names across tensors → sizes must match.
 
-- Reduction-dim validation: do NOT silently wrap out-of-range indices with `% x.ndim`. Canonical predicates and value extractors live in `tileops.manifest.shape_rules` and are exposed as shape_rule builtins (callable by bare name from any rule body, alongside `broadcast_shapes` etc.); new reduction ops MUST reference them, inline string expressions are a transitional fallback only.
+- `shape_rules` are Python expressions describing shape relationships.
 
-- `status` is required: one of `implemented` or `spec-only`.
+- Reduction-dim validation: never silently wrap out-of-range indices with `% x.ndim`. Use the canonical predicates / extractors in `tileops.manifest.shape_rules` (callable by bare name from any rule body); inline string expressions are a transitional fallback only.
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/domain-rules/manifest-spec.md]: Applied: folded the reduction-dim validation guidance into the shape_rules bullet (it's a shape_rules concern).

File: .claude/domain-rules/manifest-spec.md [hunk 8]
```diff
@@ -8,42 +8,42 @@
...
+- Never modify manifest to match non-conforming code. Code drift → `status: spec-only` and fix code in a follow-up PR. Never remove `params`, roofline `vars`, or `shape_rules` to silence validator errors.
 
-- **Manifest comment policy.** Manifest YAML carries technical content the DSL cannot express structurally (schema clarifications, edge cases, conventions, file-level headers). It does **not** carry development-process metadata — anything bound to a specific issue, PR, commit, or development round. Test: would this comment still be meaningful if every issue / PR had different numbers and every milestone was renamed? Yes → keep. No → move to commit message, PR description, or follow-up issue.
+- **Manifest comment policy.** Comments may carry technical content the DSL can't express (schema clarifications, edge cases, conventions, file headers). They MUST NOT carry development-process metadata bound to a specific issue / PR / commit / round. Test: would the comment still be meaningful if every issue and PR were renumbered? Yes → keep. No → move to commit message, PR description, or follow-up issue.
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/domain-rules/manifest-spec.md]: Applied: dropped the 'Test:' phrasing in the manifest comment policy; kept the renumbering test as the directive's logic.

File: .claude/domain-rules/ops-design.md [hunk 9]
```diff
@@ -7,10 +7,10 @@
...
-- For arbitrary-rank ops, use `static_dims` in the manifest to declare values the user commits to at construction time. Each entry is a single-axis reference `<tensor>.shape[<const_or_param>]`. Dimensions not in `static_dims` are derived from tensors at forward time. See [manifest.md R20](../../docs/design/manifest.md).
-- When adding or modifying an intermediate base class, changing kernel dispatch patterns, or introducing new class variable protocols, update `docs/design/ops-design.md` to reflect the change.
-- When adding a new op family that inherits `Op` directly, evaluate whether it shares `forward()` flow with an existing family before creating a new base class. Document the decision in the PR.
+- Op class names: PascalCase `{Name}{Direction}Op`, direction suffix mandatory. Manifest author chooses `{Name}`.
+- Kernel class names: PascalCase `{Name}{Direction}Kernel`. Builder functions stay snake_case.
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/domain-rules/ops-design.md]: Applied: merged Op and Kernel naming conventions into a single class-name directive.

File: .claude/rules/code-style.md [hunk 10]
```diff
@@ -18,21 +18,16 @@
...
   ```
 
-  Cleanup must describe the invariant to restore, not reference a specific PR. Scan with `grep -rn 'FIXME(staged-rollout)'`.
+- **Abbreviation casing in PascalCase**: standard abbreviations stay fully uppercase — `RMSNormKernel`, `SSDDecodeOp`, `FusedAddRMSNormFwdOp`.
 
-- **Abbreviation casing in PascalCase symbols**: Standard abbreviations must be fully uppercase — `RMS`, not `Rms`; `SSD`, not `Ssd`; `SSM`, not `Ssm`. Examples: `RMSNormKernel`, `SSDDecodeOp`, `FusedAddRMSNormFwdOp`.
+- **Filenames**: all-lowercase with underscores; multi-letter abbreviations stay lowercase — `rms_norm.py`, `ssd_decode.py`. Don't capitalize a single letter.
 
-- **Abbreviation casing in filenames**: Filenames use all-lowercase with underscores. Multi-word abbreviations keep all letters lowercase — `rms_norm.py`, `ssd_decode.py`. Do not capitalize a single letter (e.g. `Ssd_decode.py` is wrong).
+- **Norm filenames**: underscore-separated — `rms_norm`, `layer_norm`, `batch_norm`, `fused_add_rms_norm`. Never contracted (`rmsnorm` etc.).
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/rules/code-style.md]: Applied: folded the norm-filename specialization into the general filenames bullet.

File: .claude/rules/code-style.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/rules/code-style.md]: Already addressed in commit d49372d before this re-review surfaced — current HEAD has the imperative 'MUST' wording / 'Validator enforces presence + string type only; semantics not checked' phrasing per your earlier comment. This thread is a duplicate of the resolved one above.

File: .claude/domain-rules/manifest-spec.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/domain-rules/manifest-spec.md]: Already addressed in commit d49372d before this re-review surfaced — current HEAD has the imperative 'MUST' wording / 'Validator enforces presence + string type only; semantics not checked' phrasing per your earlier comment. This thread is a duplicate of the resolved one above.

File: .claude/domain-rules/manifest-spec.md [hunk 11]
```diff
@@ -8,42 +8,34 @@
 
 ______________________________________________________________________
 
-- Manifest key must equal the corresponding Op `cls.__name__` exactly. Class-naming convention is in `ops-design.md`.
+- Manifest key must equal the Op `cls.__name__` exactly.
 
-- `ref_api` (required) — declares the external API the signature follows (e.g., `torch.nn.functional.rms_norm`). Set to `"none"` if no direct external counterpart exists. Informational only; does not affect validation.
+- `ref_api` (required): the external API the signature mirrors (e.g. `torch.nn.functional.rms_norm`); `"none"` if none. Validator enforces presence + string type only; semantics not checked.
```
[the remark this answers is not in the document]
Comment (lcy-seso) [reply .claude/domain-rules/manifest-spec.md]: Restored: appended 'Class-naming convention: see ops-design.md' to the manifest-key bullet. The previous bullet-merge collapsed the pointer; this brings it back as a minimal clause.

**Review by Ibuki-wind — APPROVED** [review]
Autonomous review: HEAD 612f13e, CI green, 0 unresolved threads. Pass 1+2 of agent-doc brevity sweep on rules/ + domain-rules/. All copilot wording fixes applied (4 across 4 commits) and gemini consolidations applied (6 medium); class-naming cross-reference restored after the bullet-merge dropped it.
